### PR TITLE
25 Unit test cases for power control

### LINKED ISSUE #25: Unit test cases for power control
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Unit test case for power control

### New or Affected Resource(s)

- resource_redfish_power.go

You can assign @grantcurell

### PR DESCRIPTION
**Pull request by grantcurell**
.gitignore:
- Added go.sum and *.zip

go.sum:
- Removed because everyone should regenerate this on their own before running

go.mod:
- Updated the version for the terraform plugin SDK

common.go:
- Ran optimize imports (which just changed the import order)

tests:
- Removed all old unit tests

resource_redfish_power.go:
- Removed NMI and PushPowerButton because we do not implement those and they aren't particularly useful

tests:
- Added an acceptance test for power

go.mod:
- Updated the version of gofish required until it is tagged appropriately in the main gofish repo

- Fixes #25

Signed-off-by: Grant Curell <grant_curell@dell.com>

### CODE REVIEW

**Comment by grantcurell** [comment]
The only thing I'm not sure how to handle just yet is how to more gracefully pass the IP address which you want to test itself

File: redfish/provider_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package redfish
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+var testAccProviders map[string]*schema.Provider
+var testAccProvider *schema.Provider
+
```
Comment (mikeletux) [inline redfish/provider_test.go:7]: Hey @grantcurell 
I propose the following. Why don't we define an struct lile:
~~~
type TestingServerCredentials struct {
	Username string
	Password string
	Endpoint string
	Insecure bool
}
~~~
And we create a var under *testAccProvider* like:
~~~
var credentials *TestingServerCredentials
~~~
So we can init that at the init() func, and grab the actual values from ENV vars to make test fit every single scenario (and also to facilitate us to integrate it in our future CI/CD pipeline)
~~~
func init() {
	testAccProvider = Provider()
	testAccProviders = map[string]*schema.Provider{
		"redfish": testAccProvider,
	}
	
	credentials = &TestingServerCredentials{
		Username: os.Getenv("TF_TESTING_USERNAME"),
		Password: os.Getenv("TF_TESTING_PASSWORD"),
		Endpoint: os.Getenv("TF_TESTING_ENDPOINT"),
		Insecure: false,
	}
} 
~~~

File: redfish/resource_redfish_power_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+					                                		"192.168.1.63",
+														 "On",
+														  120,
+												  				10),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("redfish_power.system_power", "power_state", "On"),
+				),
+			},
+			{
+				Config: testAccRedfishResourcePowerConfig(
```
Comment (mikeletux) [inline redfish/resource_redfish_power_test.go:28]: By using what I said in the other comment, we would avoid hardcoding values around (credentials related) and we could only pass the ones regarding resources.

**Comment by grantcurell** [comment]
@mikeletux all requested changes made and tested.

File: redfish/provider_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// TestingServerCredentials Struct used to store the credentials we pass for testing. This allows us to pass testing
+// credentials via environment variables instead of having them hard coded
+type TestingServerCredentials struct {
+	Username string
+	Password string
+	Endpoint string
+	Insecure bool
+}
+
+var creds = TestingServerCredentials{
```
Comment (mikeletux) [inline redfish/provider_test.go:20]: Hi @grantcurell,

Just to make it compliant with the other two vars, I think you should declare the *creds* var under *testAccProviders* and *testAccProvider* as an empty var, and then init it within the *init()* function, as those two vars are:
Something like:
~~~
var testAccProviders map[string]*schema.Provider
var testAccProvider *schema.Provider
var creds TestingServerCredentials

func init() {

	testAccProvider = Provider()
	testAccProviders = map[string]*schema.Provider{
		"redfish": testAccProvider,
	}
        
        creds = TestingServerCredentials{
                           Username: os.Getenv("TF_TESTING_USERNAME"),
	                   Password: os.Getenv("TF_TESTING_PASSWORD"),
	                   Endpoint: os.Getenv("TF_TESTING_ENDPOINT"),
	                   Insecure: false,
                   }
}
~~~

Everyting else look good to me 😄

Comment (grantcurell) [reply redfish/provider_test.go:20]: @mikeletux I tried that - there's something about the scope that doesn't work:

image

**Pull request merged**